Tools/build/check_warnings.py

```diff
@@ -4,9 +4,9 @@
 """
 
 import argparse
-from collections import defaultdict
 import re
 import sys
+from collections import defaultdict
```

Comment:
If you make the following change, this `defaultdict` import can be removed:

```diff
-warnings_by_file = defaultdict(list)
+warnings_by_file = {}
```

```diff
-warnings_by_file[warning["file"]].append(warning)
+warnings_by_file.setdefault(warning["file"], []).append(warning)
```

```suggestion
```

Reply:
I generally prefer `defaultdict`; it's more concise at the point of use and faster. But I don't have a strong opinion :-)